bundles/org.openhab.ui/web/src/components/item/item-form.vue

```diff
@@ -1,48 +1,81 @@
+          <option v-for="t in types.ItemTypes" :key="t">
             {{ t }}
           </option>
-        </select>
-      </f7-list-item>
-      <f7-list-item v-if="dimensions.length && item.type && !hideType && item.type.startsWith('Number')" title="Dimension" type="text" :disabled="!editable" smart-select :smart-select-params="{searchbar: true, openIn: 'popup', closeOnSelect: true}">
-        <select name="select-dimension" @change="setDimension($event.target.value)">
-          <option key="Number" value="Number" :selected="item.type === 'Number'" />
-          <option v-for="(d, i) in dimensions" :key="d.name" :value="i" :selected="'Number:' + d.name === item.type">
+        </f7-list-input>
+
+        <!-- Dimensions -->
+        <f7-list-input v-if="dimensions.length && item.type.startsWith('Number')"
+                       label="Dimension"
+                       type="select"
+                       :disabled="!editable"
+                       :value="itemDimension"
+                       @input="setDimension($event.target.value)">
+          <option value="" />
+          <option v-for="d in dimensions" :key="d.name" :value="d.name">
             {{ d.label }}
           </option>
-        </select>
-      </f7-list-item>
-      <!-- Use v-show instead of v-if, because otherwise the autocomplete for category would take over the unit -->
-      <f7-list-input v-show="!hideType && item.type && item.type.startsWith('Number:') && createMode" label="Unit" type="text" :value="item.unit"
-                     info="Used internally, for persistence and external systems. It is independent from the state visualization in the UI, which is defined through the state description."
-                     @input="item.unit = $event.target.value" :disabled="!editable" :clear-button="editable" />
-      <f7-list-input v-show="!hideType && item.type && item.type.startsWith('Number:') && createMode" label="State Description Pattern" type="text" :value="item.stateDescriptionPattern"
-                     info="Pattern or transformation applied to the state for display purposes. Only saved if you change the pre-filled default value."
-                     @input="item.stateDescriptionPattern = $event.target.value" :disabled="!editable" :clear-button="editable" />
-      <f7-list-input v-if="!hideCategory" ref="category" label="Category" autocomplete="off" type="text" placeholder="temperature, firstfloor..." :value="item.category"
-                     @input="item.category = $event.target.value" :disabled="!editable" :clear-button="editable">
-        <div slot="root-end" style="margin-left: calc(35% + 8px)">
-          <oh-icon :icon="item.category" :state="(createMode) ? null : item.state" height="32" width="32" />
-        </div>
-      </f7-list-input>
+        </f7-list-input>
+        <!-- Use v-show instead of v-if, because otherwise the autocomplete for category would take over the unit -->
+        <f7-list-input v-show="itemDimension"
```

Comment:
> Keeping the current behaviour would be easier, but is less user-friendly.

IMHO these fields should be editable. I said the same in @mherwege 's PR. Otherwise if you make a mistake when creating the item, the only solution is to delete and recreate again.

Reply:
That's a point for type and dimension.

For unit and state description, they still are editable for existing Items, but you currently have to edit them through the metadata. We can support them trough the Item form, but that would require some extra work. The current state of this PR is buggy with regards to that.

Reply:
> current state of this PR is buggy with regards to that.

That's why I was suggesting to follow @mherwege 's PR..

Reply:
In that case I would prefer to follow my suggested changes and continue to hide those fields with this PR here, and then take care of that in #2312.

Reply:
I guess it depends whether #2312 or #2326 will be merged first. Either way , the merge conflict needs to be clearly resolved.

Reply:
As I’m finished with the review of this PR and waiting for it being addressed, I guess this PR will be merged first.

I’ll try to have a look at #2312 soon but can’t promise that I’ll have the time.

Reply:
Indeed the unit and state description is currently buggy - it only works during item creation but doesn't work during item editing. So thanks for catching that! 

The issue lies in item-mixin's `saveItem()` which only saves them during createMode. 

If we allow changing the dimension during item editing, wouldn't it make a lot more sense to allow changing the state description and unit? Otherwise, when the dimension or even the type is changed, the item would end up with an entirely wrong unit.  

Currently:
- Create a new item, set to Number:Power. The default unit is W
- Set unit to kW, State Description to %.1f kW
- Save
- Now edit the item, change the dimension to Energy or change the Type to String
- Save
- unit and State Description stays as kW and %.1f kW

This to me is a bug, but perhaps fixing this should be done in a separate PR

In any case, this seems like fixing this should belong in a separate PR because otherwise it's going to make this PR even bigger. So for that reason, I will remove the change in this PR, and revert the behaviour to only show the unit and state description during item creation as it was before the PR.

Reply:
> IMO is questionable if we should allow changing the dimension or the Item type at all, since this will likely break existing channel links and rules etc.

PS I don't agree with your opinion. While it is indeed possible that changing these attributes might indeed break already working rules etc., it is equally likely that you might have a rule that is not working (due to initial wrong choice of unit or dimension) and you need to edit those attributes in order to fix the problem.

Your current proposal of hiding those attributes would a) not even allow you to see if something was wrong, b) prevent you from fixing it, and c) therefore require you to delete the item and recreate it (on a trial and error basis) until you would has fixed whatever issue there might have been.

Reply:
> So for that reason, I will remove the change in this PR, and revert the behaviour to only show the unit and state description during item creation as it was before the PR.

:+1:

> PS I don't agree with your opinion.

You have strong arguments. Let's keep it as is and probably introduce a warning popup that asks for confirmation when changing the Item type or dimension or unit in a different PR.